recipes/qt/6.x.x/conanfile.py

```diff
@@ -522,26 +526,137 @@ def package_id(self):
                 self.info.settings.compiler.runtime = "MT/MTd"
 
     def package_info(self):
-        # FIXME add components
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package_multi"] = "Qt6"
 
-        # Add top level include directory, so code compile if someone uses
-        # includes with prefixes (e.g. "#include <QtCore/QString>")
-        self.cpp_info.includedirs = ["include"]
+        libsuffix = ""
+        if self.settings.build_type == "Debug":
+            if self.settings.os == "Windows":
+                libsuffix = "d"
+            if tools.is_apple_os(self.settings.os):
+                libsuffix = "_debug"
+
+        def _create_module(module, requires=[]):
+            assert module not in self.cpp_info.components, "Module %s already present in self.cpp_info.components" % module
+            self.cpp_info.components[module].names["cmake_find_package"] = module
```

Comment:
module values passed to this function lead to too generic component names, it might lead to side effects with pkg_config generator